www/src/pages/community/companies.js

```diff
@@ -0,0 +1,41 @@
+import React, { Component } from "react"
+import { graphql } from "gatsby"
+import CommunityView from "../../views/community"
+
+class CompaniesPage extends Component {
+  render() {
+    const { location, data } = this.props
+    return <CommunityView data={data} location={location} title={`Companies`} />
+  }
+}
+
+export default CompaniesPage
+
+export const pageQuery = graphql`
+  query {
+    allCreatorsYaml(filter: { type: { eq: "company" } }) {
+      edges {
+        node {
+          name
+          description
+          location
+          website
+          github
+          image {
+            childImageSharp {
+              fixed(width: 240, height: 240) {
+                ...GatsbyImageSharpFixed
+              }
+            }
+          }
+          hiring
+          for_hire
+          portfolio
+          fields {
+            slug
+          }
+        }
+      }
```

Comment:
Basically my thinking was these are sub navigation items overall, and main navigation items on the community page itself. If we don't use direct links via <Link to={}> and instead resolve from gatsby-node.js as a template, we need to do checks to see where we are, send that as a filter to graphql query,  set that for slugs and use programmatic navigation via other means instead of direct links? 

This seemed the easiest (and clearer/less "magical" for anyone who would look into the navigation code) way of creating these routes to me. I agree about the repeated code and @DSchau 's fragment suggestion seems feasible for the time being. 

I will fix the other issues first and will circle around to this after creating the fragments, also if I recall correctly, something did not work when I was trying to make these work via gatsby-node, but can't remember what it was now 😆

Reply:
I think the graphql fragment suggested by @DSchau below should get rid of the bulk of the duplication.